openghg/standardise/column/_openghg.py

```diff
@@ -0,0 +1,315 @@
+from typing import cast, Union, Optional, Dict, List, MutableMapping
+from pathlib import Path
+import xarray as xr
+
+
+def parse_openghg(
+    data_filepath: Union[str, Path],
+    satellite: Optional[str] = None,
+    domain: Optional[str] = None,
+    selection: Optional[str] = None,
+    site: Optional[str] = None,
+    species: Optional[str] = None,
+    network: Optional[str] = None,
+    instrument: Optional[str] = None,
+    platform: str = "satellite",
+    **kwargs: str,
+) -> Dict:
+    """
+    Parse and extract data from pre-formatted netcdf file which already
+    matches expected OpenGHG format.
+
+    The arguments specified below are the metadata needed to store this
+    surface observation file within the object store. If these keywords are
+    not included within the attributes of the netcdf file being passed then
+    these arguments must be specified.
+
+    For column data this can either be a satellite (e.g. satellite="GOSAT") or a
+    site (site="RUN", network="TCCON"). Either can be specified or this function
+    will attempt to extract this from the data file.
+
+    Args:
+        data_filepath: Path of observation file
+        satellite: Name of satellite (if relevant)
+        domain: For satellite only. If data has been selected on an area include the
+            identifier name for domain covered. This can map to previously defined domains
+            (see domain_info.json) or a newly defined domain.
```

Comment:
Maybe we could have a helper function that shows the domains? If people have this installed it might not be that easy to find the JSON.

Reply:
See Issue #393